providers/ldap/ldap_client.go

```diff
@@ -783,12 +784,21 @@ func (l *LClient) userRecord(search *ldap.SearchRequest, lConn *ldap.Conn) (mode
 		return nilToken, status, err
 	}
 
+	err = nil
 	if len(result.Entries) < 1 {
 		log.Errorf("Cannot locate user information for %s", search.Filter)
-		return nilToken, status, nil
+		if l.AccessMode == "required" {
+			status = 403
+			err = errors.New("Account does not have access")
+		}
+		return nilToken, status, err
```

Comment:
I would not generalize to "Account does not have access" in the common method userRecord(). We should do throw this error in the caller code. userRecord() is called by refreshToken and can be used in future methods - so  we should handle throwing this error only in the caller where accessmode check is applied.